docs/usage.md

```diff
@@ -70,6 +45,76 @@ INIT_RUNTIME(2); // This is the number of pkeys needed
 
 Note that this must only be included in one source file per DSO.
 
+### Capturing Compile Commands
+
+The rewriter relies on a `compile_commands.json` file with information about the
+compiler flags used to compile each source file. If your build system does not
+directly support generating a compile commands file, you can use a tool like
+`bear` to record the compiler invocations that your build system makes.
+
+One limitiation is that the rewriter currently does not support having multiple
+compile commands for the same source file. If your build system builds the same
+source file multiple times, e.g. building for both a shared lib and a static
+lib, you'll need to change your build such that it only builds the sources once.
+
+Note that `libclangTooling`, which `ia2-rewriter` uses, is very finicky
+about relative paths, so we suggest making all relative paths absolute.
+Furthermore, `ia2-rewriter` itself sometimes uses paths as keys,
+so we suggest making all paths canonical. We have a script,
+[`canonicalize_compile_command_paths.py`](../tools/rewriter/canonicalize_compile_command_paths.py),
+that does this automatically (doing its best to detect paths embedded in args).
+`libclangTooling` also requires compilation databases to be named exactly
+`compile_commands.json`, so this script assumes that as well,
+and must be run in the directory of the `compile_commands.json`, like this:
+
+```sh
+(cd $build_directory && $ia2_dir/tools/rewriter/canonicalize_compile_command_paths.py)
+```
+
+### Running the Rewriter
```

Comment:
Does it make sense to call it this when this is inside of `## Running the Rewriter`?

Reply:
Oh lol that's a good point. I'll change the first header to "Setup and Rewriter".